### PR TITLE
Add touch-pad support

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/gpio/touch.toit [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+Use the $Touch.threshold (als settable during construction) to define at which level the
+  touch pad should be considered "touched". Typically, a calibration round reads the
+  touch pad value (multiple times) when the program starts and sets the threshold to 2/3 of
+  the average value.
+
+A touch pad can be used to wake the device from deep sleep. See the 'esp32' library.
+*/
+
+/**
+Touch Sensor.
```
Comment (kasperl) [inline lib/gpio/touch.toit:34]:
```suggestion
Touch sensor.
```

File: src/resources/touch_esp32.cc [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    case TOUCH_PAD_NUM8: return 33;
+    default: return -1;
+  }
+}
+
+#elif CONFIG_IDF_TARGET_ESP32S2
+
+static touch_pad_t get_touch_pad(int pin) {
+  switch (pin) {
+    case 1: return TOUCH_PAD_NUM1;
```
Comment (kasperl) [inline src/resources/touch_esp32.cc:76]: Want to align the returns like in touch_pad_to_pin_num?

File: src/resources/touch_esp32.cc [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    case TOUCH_PAD_NUM13: return 13;
+    case TOUCH_PAD_NUM14: return 14;
+    default: return -1;
+  }
+}
+
+#elif CONFIG_IDF_TARGET_ESP32C3
+
+static touch_pad_t get_touch_pad(int pin) {
+  // Esp32c3 does not have touch support.
```
Comment (kasperl) [inline src/resources/touch_esp32.cc:117]:
```suggestion
  // ESP32-C3 does not have touch support.
```

File: src/resources/touch_esp32.cc [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+  return touch_pad_t(-1);
+}
+
+int touch_pad_to_pin_num(touch_pad_t pad) {
+  // Esp32c3 does not have touch support.
```
Comment (kasperl) [inline src/resources/touch_esp32.cc:122]:
```suggestion
  // ESP32-C3 does not have touch support.
```

File: src/primitive_esp32.cc [hunk 5]
```diff
@@ -310,6 +311,23 @@ PRIMITIVE(enable_external_wakeup) {
...
+  if (err != ESP_OK) {
+    ESP_LOGE("Toit", "Failed: sleep_enable_touchpad_wakeup");
+    OTHER_ERROR;
+  }
+  err = esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
+  if (err != ESP_OK) {
+    ESP_LOGE("Toit", "Failed: sleep_enable_touchpad_wakeup - power domain");
+    OTHER_ERROR;
+  }
+  keep_touch_active();
```
Comment (kasperl) [inline src/primitive_esp32.cc:326]: There is no way to disable this again?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/primitive_esp32.cc (hunk 5 above)
Comment (floitsch) [reply src/primitive_esp32.cc:326]: Not at the moment. Once you enabled wakeup from deepsleep with touch-pins you can't disable it again.
We could add a function to `esp32` to disable wakeups again.
We don't have it for gpio either.

File: lib/gpio/touch.toit (hunk 1 above)
Comment (floitsch) [reply lib/gpio/touch.toit]: done

File: src/resources/touch_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/touch_esp32.cc]: done.

File: src/resources/touch_esp32.cc (hunk 3 above)
Comment (floitsch) [reply src/resources/touch_esp32.cc]: done.

File: src/resources/touch_esp32.cc (hunk 4 above)
Comment (floitsch) [reply src/resources/touch_esp32.cc]: done.

**Pull request merged**